src/graph.cpp

```diff
@@ -1,12 +1,12 @@
-#include "../include/graph.h"
-
-void Graph::addNode(int node) {
-    if (adjList.find(node) == adjList.end()) {
-        adjList[node] = std::vector<std::pair<int, int>>();
-        this->size++;
-    }
-}
-
-void Graph::addEdge(int src, int dest, int weight) {
-    adjList[src].push_back({dest, weight});
-}
```

Comment:
you're missing graceful error handling for when your arguments are unsound (e.g. index out of bounds)